test/e2e/wait_timeout_test.go

```diff
@@ -0,0 +1,261 @@
+kind: Deployment
+metadata:
+  namespace: ns3
+  name: simple-app3
+spec:
+  selector:
+    matchLabels:
+      simple-app3: ""
+  template:
+    metadata:
+      labels:
+        simple-app3: ""
+    spec:
+      containers:
+        - name: simple-app3
+          image: docker.io/dkalinin/k8s-simple-app@sha256:4c8b96d4fffdfae29258d94a22ae4ad1fe36139d47288b8960d9958d1e63a9d0
+          env:
+            - name: HELLO_MSG
+              value: stranger
+`
+
+	name := "test-wait-timeout"
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+	}
+
+	cleanUp()
+	defer cleanUp()
+
+	logger.Section("Resource wait timeout", func() {
+		out, _ := kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name, "--wait-timeout",
+			"2m", "--wait-resource-timeout", "2ms", "--json"},
+			RunOpts{IntoNs: true, AllowError: true, StdinReader: strings.NewReader(yaml1)})
+
+		expected := []map[string]string{{
+			"conditions":      "",
+			"kind":            "Namespace",
+			"name":            "ns1",
+			"namespace":       "(cluster)",
+			"op":              "create",
+			"op_strategy":     "",
+			"reconcile_info":  "",
+			"reconcile_state": "",
+			"wait_to":         "reconcile",
+		}, {
+			"conditions":      "",
+			"kind":            "Namespace",
+			"name":            "ns2",
+			"namespace":       "(cluster)",
+			"op":              "create",
+			"op_strategy":     "",
+			"reconcile_info":  "",
+			"reconcile_state": "",
+			"wait_to":         "reconcile",
+		}, {
+			"conditions":      "",
+			"kind":            "Namespace",
+			"name":            "ns3",
+			"namespace":       "(cluster)",
+			"op":              "create",
+			"op_strategy":     "",
+			"reconcile_info":  "",
+			"reconcile_state": "",
+			"wait_to":         "reconcile",
+		}, {
+			"conditions":      "",
+			"kind":            "Deployment",
+			"name":            "simple-app",
+			"namespace":       "kapp-test",
+			"op":              "create",
+			"op_strategy":     "",
+			"reconcile_info":  "",
+			"reconcile_state": "",
+			"wait_to":         "reconcile",
+		}, {
+			"conditions":      "",
+			"kind":            "Deployment",
+			"name":            "simple-app2",
+			"namespace":       "kapp-test",
+			"op":              "create",
+			"op_strategy":     "",
+			"reconcile_info":  "",
+			"reconcile_state": "",
+			"wait_to":         "reconcile",
+		}, {
+			"conditions":      "",
+			"kind":            "Deployment",
+			"name":            "simple-app3",
+			"namespace":       "kapp-test",
+			"op":              "create",
+			"op_strategy":     "",
+			"reconcile_info":  "",
+			"reconcile_state": "",
+			"wait_to":         "reconcile",
+		}}
+
+		resp := uitest.JSONUIFromBytes(t, []byte(out))
+
+		validateChanges(t, resp.Tables, expected, "Op:      6 create, 0 delete, 0 update, 0 noop",
+			"Wait to: 6 reconcile, 0 delete, 0 noop", out)
+
+		if len(resp.Lines) > 0 && !strings.Contains(resp.Lines[len(resp.Lines)-1] ,"Resource timed out waiting after 2ms") {
```

Comment:
instead of looking at lines lets check err.Error() contains that string.

Reply:
Yeah Done